### DISCUSSION TITLE
How to ensure video transmission quality when network congestion occurs? Can I use libdatachannel library to reduce the bitrate/resolution/frame rate of the video, or is there a better solution?

### DISCUSSION THREAD

**Discussion by zhouizhan** [paullouisageneau/libdatachannel#1307] [Q&A] [answered]
thank you！

**Comment by paullouisageneau** [comment, marked as answer]
For media transport, libdatachannel does not encode so it can do nothing by itself, you need to use a realtime encoder allowing you to change the bitrate dynamically. There is currently no rate controller shipped with the library but you can implement your own with media interceptors, for instance @vagonhq has a [fork with a rate controller](https://github.com/vagonhq/libdatachannel/tree/twcc-v0.21.2).
